impeller/toolkit/android/toolkit_android_unittests.cc

```diff
@@ -0,0 +1,63 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "flutter/fml/synchronization/waitable_event.h"
+#include "flutter/testing/testing.h"
+#include "impeller/toolkit/android/hardware_buffer.h"
+#include "impeller/toolkit/android/proc_table.h"
+#include "impeller/toolkit/android/surface_control.h"
+#include "impeller/toolkit/android/surface_transaction.h"
+
+namespace impeller::android::testing {
+
+class ToolkitAndroidTest : public ::testing::Test {
+ public:
+  void SetUp() override {
+    // The toolkit is only available on Android API levels over 29. Skip these
+    // tests everywhere else.
+    if (__builtin_available(android 29, *)) {
+    } else {
```

Comment:
nit: This might read a little clearer and avoid a builtin via

```suggestion
    if (android_get_device_api_level() < 29) {
```

Reply:
I, uh, didn't want to link against `libAndroid.so`.

Reply:
Oh, that's not why. `android_get_device_api_level` is itself only available on API level 29. So you can't call it on lower API levels. The act of checking if you are on a lower API level is itself unsafe.